### ISSUE TITLE
Add support for mini-batch training

### ISSUE THREAD

**Issue by pat-alt** [JuliaTrustworthyAI/LaplaceRedux.jl#19]
Currently the Hessian is approximated per data point,

```julia
# Training:
for d in data
    loss_batch, H_batch = hessian_approximation(la, d)
    loss += loss_batch
    H += H_batch
    n_data += 1
end
```

where d is a tuple of inputs and outputs `(x,y)` at instance level. For various reasons it would be better to add support for training in batches.

**Comment by pat-alt** [comment]
Hinges on #20

**Comment by pat-alt** [comment]
closed by #33
